hstrat/_auxiliary_lib/_bit_ceil.py

```diff
@@ -2,7 +2,8 @@ def bit_ceil(n: int) -> int:
     """Calculate the smallest power of 2 not smaller than n."""
     if n:
         # see https://stackoverflow.com/a/14267825/17332200
-        exp = (n - 1).bit_length()
+        # cast to int to make robust to numpy.int32, numpy.int64, etc.
+        exp = int(n - 1).bit_length()
```

Comment:
Do we need to bounds check for n < 0?

Reply:
checked & this definitely doesn't work for negative integers. Adding an assert and a docstring note!